python/ray/_private/structured_logging/logging_config.py

```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+
+from ray._private.structured_logging.constants import LOG_MODE_DICT
+from ray.util.annotations import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+class LoggingConfig:
+    def __init__(self, encoding: Optional[str] = "TEXT", log_level: str = "INFO"):
+        """
+        The class is used to store the Python logging configuration. It will be applied
+        to the root loggers of the driver process and all Ray task and actor processes
+        that belong to the this job.
+
+        Examples:
+            .. testcode::
+
+                import ray
+                import logging
+
+                ray.init(
+                    logging_config=ray.LoggingConfig(encoding="TEXT", log_level="INFO")
+                )
+
+                @ray.remote
+                def f():
+                    logger = logging.getLogger()
+                    logger.info("This is a Ray task")
+
+                obj_ref = f.remote()
+                ray.get(obj_ref)
+
+        Args:
+            encoding: encoding is a string, and it should be one of the keys in
+                LOG_MODE_DICT, which has the corresponding predefined logging
+                configuration.
```

Comment:
"Encoding type for the logs. The valid value is 'TEXT'".

Ideally we can auto generate the docstring based on the value of LOG_MODE_DICT by doing 

```
LoggingConfig.__doc__ = f"""
...
....
{str(LOG_MODE_DICT)}
"""
```

Reply:
> "Encoding type for the logs. The valid value is 'TEXT'".

Updated https://github.com/ray-project/ray/pull/45344/commits/c8621a0476b43870f936da4908a64f3e8cb052e4

> Ideally we can auto generate the docstring based on the value of LOG_MODE_DICT by doing

TODO

Reply:
> LoggingConfig.__doc__ = f"""

Do we need to include this in this PR?